### PR TITLE
Add method for updating a pull request review

### PR DESCRIPTION
**Pull request by joshuabezaleel**
Fixes: #1174 

- Add UpdateReview to PullRequestReview.
- Add its test.

### CODE REVIEW

File: github/pulls_reviews.go [hunk 1]
```diff
@@ -189,6 +189,29 @@ func (s *PullRequestsService) CreateReview(ctx context.Context, owner, repo stri
...
+// UpdateReview updates the review summary on the specified pull request.
+//
+// GitHub API docs: https://developer.github.com/v3/pulls/reviews/#update-a-pull-request-review
+func (s *PullRequestsService) UpdateReview(ctx context.Context, owner, repo string, number int, reviewID int64, body string) (*PullRequestReview, *Response, error) {
+	opts := &struct {
+		Body string `json:"body"`
+	}{Body: body}
+	u := fmt.Sprintf("repos/%v/%v/pulls/%d/reviews/%d", owner, repo, number, reviewID)
+
+	req, err := s.client.NewRequest("PATCH", u, opts)
```
Comment (gmlewis) [inline github/pulls_reviews.go]: This needs to be `PUT`, not `PATCH`.

File: github/pulls_reviews_test.go [hunk 2]
```diff
@@ -195,6 +195,26 @@ func TestPullRequestsService_CreateReview_invalidOwner(t *testing.T) {
 	testURLParseError(t, err)
 }
 
+func TestPullRequestsService_UpdateReview(t *testing.T) {
+	client, mux, _, teardown := setup()
+	defer teardown()
+
+	mux.HandleFunc("/repos/o/r/pulls/1/reviews/1", func(w http.ResponseWriter, r *http.Request) {
+		testMethod(t, r, "PATCH")
```
Comment (gmlewis) [inline github/pulls_reviews_test.go]: s/PATCH/PUT/

File: go.sum [hunk 3]
```diff
@@ -1,5 +1,6 @@
 github.com/golang/protobuf v1.2.0 h1:P3YflyNX/ehuJFLhxviNdFxQPkGK5cDcApsge1SqnvM=
 github.com/golang/protobuf v1.2.0/go.mod h1:6lQm79b+lXiMfvg/cZm0SGofjICqVBUtrP5yJMmIC1U=
+github.com/google/go-github v17.0.0+incompatible h1:N0LgJ1j65A7kfXrZnUDaYCs/Sf4rEjNlfyDHW9dolSY=
```
Comment (gmlewis) [inline go.sum]: Please revert the contents of this file in this PR.

File: github/pulls_reviews_test.go (hunk 2 above)
Comment (joshuabezaleel) [reply github/pulls_reviews_test.go]: I am really sorry if I do not get this. Where should we put the `s/PATCH/PUT`?

Comment (gmlewis) [reply github/pulls_reviews_test.go]: I apologize... that's shorthand for "substitute 'PATCH' with 'PUT'" (It's a vi/vim/sed [substitute command](https://vim.fandom.com/wiki/Search_and_replace)).
This is the same case as the comment above for `github/pulls_reviews.go`.

**Comment by gmlewis** [comment]
Thank you, @gauntface!
Merging.

**Comment by joshuabezaleel** [comment]
Thank you very much, @gmlewis and @gauntface ! :)

**Pull request merged**